[Merge to M63] Linux sandbox: Fix GetGlibcCachedTid returning 0 on some systems

> BUG=776359
> R=jorgelo@chromium.org
>
> Change-Id: Ic792ffc039ce32e8199b6dc93ff6e5e6dac92fab
> Reviewed-on: https://chromium-review.googlesource.com/732483
> Reviewed-by: Jorge Lucangeli Obes <jorgelo@chromium.org>
> Commit-Queue: Thomas Anderson <thomasanderson@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#510855}

BUG=776359
TBR=jorgelo@chromium.org
NOTRY=true
NOPRESUBMIT=true
NOTREECHECKS=true

Change-Id: Ic792ffc039ce32e8199b6dc93ff6e5e6dac92fab
Reviewed-on: https://chromium-review.googlesource.com/732483
Reviewed-by: Jorge Lucangeli Obes <jorgelo@chromium.org>
Commit-Queue: Thomas Anderson <thomasanderson@chromium.org>
Cr-Original-Commit-Position: refs/heads/master@{#510855}
Reviewed-on: https://chromium-review.googlesource.com/735212
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Cr-Commit-Position: refs/branch-heads/3239@{#172}
Cr-Branched-From: adb61db19020ed8ecee5e91b1a0ea4c924ae2988-refs/heads/master@{#508578}

diff --git a/sandbox/linux/services/namespace_sandbox.cc b/sandbox/linux/services/namespace_sandbox.cc
index 3e5bba3..53d530c 100644
--- a/sandbox/linux/services/namespace_sandbox.cc
+++ b/sandbox/linux/services/namespace_sandbox.cc
@@ -105,8 +105,7 @@
 };
 
 pid_t GetGlibcCachedTid() {
-  pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
-  CHECK_EQ(0, pthread_mutex_init(&lock, nullptr));
+  pthread_mutex_t lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
   CHECK_EQ(0, pthread_mutex_lock(&lock));
   pid_t tid = lock.__data.__owner;
   CHECK_EQ(0, pthread_mutex_unlock(&lock));
